src/solana/utils/omlpService.ts

```
@@ -0,0 +1,323 @@
+    try {
+      const pools: PoolData[] = [];
+      
+      // Process each token mint to fetch its pool data
+      for (const [tokenSymbol, tokenMint] of Object.entries(TOKEN_MINTS)) {
+        try {
+          const [poolAddress] = derivePoolAddress(tokenMint);
+          
+          // Fetch the pool account data
+          const accountInfo = await this.connection.getAccountInfo(poolAddress);
+          
+          if (accountInfo && accountInfo.data) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)